cliquet/tests/support.py

```diff
@@ -86,25 +86,29 @@ class BaseWebTest(FakeAuthentMixin):
 
     api_prefix = "v0"
 
-    def get_test_app(self):
-        return webtest.TestApp(testapp(self.get_app_settings()))
+    def get_test_app(self, settings=None):
+        app = webtest.TestApp(testapp(self.get_app_settings(settings)))
+        app.RequestClass = get_request_class(self.api_prefix)
+        return app
```

Comment:
patch.dict() works in some cases but not in others.

For instance, if the settings are read at application startup, you'll need to create a new application, because patching will not work (the code you want to change the behaviour would have been called already).

I believe we don't want to remove the code I changed here in #192. I commented there.